geo/services/Geo_LocationService.php

```diff
@@ -3,163 +3,63 @@
-
-        if (!$response->isSuccessful()) {
-            return array();
-        }
-
-        $data = json_decode($response->getBody());
-        if (property_exists($data, "type") && $data->type === "error") {
-            return array();
-        }
-
-        if(isset($data->subdivisions[0])){
-            $regionName = $data->subdivisions[0]->names->en;
-        }else{
-            $regionName = "";
-        }
-
-        if(isset($data->city)){
-            $city = $data->city->names->en;
-        }else{
-            $city = "";
-        }
-
-        $data = array(
-            "ip"=>$data->traits->ip_address,
-            "country_code"=>$data->country->iso_code,
-            "country_name"=>$data->country->names->en,
-            "region_name"=>$regionName,
-            // Yes i know, i am not getting postcode etc yet.
-            "city"=>$city,
-            "latitude"=>$data->location->latitude,
-            "longitude"=>$data->location->longitude,
-            "cached"=>false
-        );
-
-        return $data;
-    }
-
-    // private function getTelizeData($ip){
-
-    //     $url = "/geip/".$ip;
-    //     $telizeClient = new \Guzzle\Http\Client("http://www.telize.com");
-    //     $response = $telizeClient->get($url, array(), array("exceptions" => false))->send();
-
-    //     if (!$response->isSuccessful()) {
-    //         return array();
-    //     }
-
-    //     $data = json_decode($response->getBody(),true);
-
-    //     $data = array(
-    //         "ip"=>$data['ip'],
-    //         "country_code"=>$data['country_code'],
-    //         "country_name"=>$data['country'],
-    //         "latitude"=>$data['latitude'],
-    //         "longitude"=>$data['longitude'],
-    //         "cached"=>false
-    //     );
-
-    //     return $data;
-    // }
-
-    private function anonymizeIp($ip) {
-        if (filter_var($ip, FILTER_VALIDATE_IP) !== false) {
-            $segments = explode('.', $ip);
-
-            switch (strlen($segments[3])) {
-                case 1:
-                case 2:
-                    // Set last segment to zero
-                    $segments[3] = 0;
-                    break;
-
-                case 3:
-                    // Keep the first digit and set the last two to zero
-                    $ending = substr($segments[3], 0, 1);
-                    if (substr($segments[3], 1) == '00') {
-                        // Set two random digits if IP already ended with two zeros
-                        $ending .= rand(0, 9);
-                        $ending .= rand(0, 9);
-                    } else {
-                        $ending .= '00';
-                    }
-                    $segments[3] = $ending;
-                    break;
-            }
-
-            $anonymizedIp = implode('.', $segments);
-            return $anonymizedIp;
-        } else {
-            // Not a valid IP
-            return $ip;
-        }
-    }
-
-}
+	public function getIpData()
```

Comment:
one improvement could be to accept an IP param but default to the requests IP (`craft()->request->getIpAddress()`)